week1/plot_trips.R

```diff
@@ -1,62 +1,205 @@
+  ggplot(aes(x = age, y = Male/Female)) +
+  geom_point() +
+  geom_smooth(method = "lm") +
+  xlim(c(18, 90)) +
+  labs(x = 'Age',
+       title = 'Male to female ratio for an age group')
+  
+
+########################################
+# plot weather data
+########################################
+# plot the minimum temperature (on the y axis) over each day (on the x axis)
+weather %>%
+  ggplot(aes(x = ymd, y = tmin)) +
+  geom_point() +
+  labs(x = 'Days',
+       y = 'Minimum temperature')
+
+# plot the minimum temperature and maximum temperature (on the y axis, with different colors) over each day (on the x axis)
+# hint: try using the pivot_longer() function for this to reshape things before plotting
+# (you can skip this and come back to it tomorrow if we haven't covered reshaping data yet)
+weather %>%
+  group_by(tmin, tmax, ymd) %>%
+  pivot_longer(c('tmin', 'tmax'), names_to = 'temperature_type', values_to = 'temperature') %>%
+  ggplot(aes(x = ymd, y = temperature, color = temperature_type)) +
+  geom_point()+
+  labs(x = 'Day',
+       title = 'minimum and maiximum temperature')
+
+########################################
+# plot trip and weather data
+########################################
+
+# join trips and weather
+trips_with_weather <- inner_join(trips, weather, by="ymd")
+
+# plot the number of trips as a function of the minimum temperature, where each point represents a day
+# you'll need to summarize the trips and join to the weather data to do this
+trips_with_weather %>%
+  group_by(ymd, tmin) %>%
+  summarize(count = n(), .groups = 'drop') %>%
+  ggplot(aes(x = tmin, y = count, label = 1:28)) +
+  labs(x = 'Minimum temperature',
+       y = 'No. of trips') +
+  geom_point() +
+  geom_text(hjust=-0.3)
+
+# repeat this, splitting results by whether there was substantial precipitation or not
+# you'll need to decide what constitutes "substantial precipitation" and create a new T/F column to indicate this
+trips_with_weather %>%
+  mutate(substantial_prcp = prcp > 0.1) %>%
+  group_by(ymd, tmin, substantial_prcp) %>%
+  summarize(count = n(), .groups = 'drop') %>%
+  ggplot(aes(x = tmin, y = count, label = 1:28)) +
+  labs(x = 'Minimum temperature',
+       y = 'No. of trips',
+       title = 'Trips tend to be low when there is substantial precipitation.') +
+  geom_point() +
+  geom_text(hjust=-0.3) +
+  facet_wrap(~substantial_prcp, ncol = 1)
+
+# add a smoothed fit on top of the previous plot, using geom_smooth
+trips_with_weather %>%
+  mutate(substantial_prcp = prcp > 0.1) %>%
+  group_by(ymd, tmin, substantial_prcp) %>%
+  summarize(count = n(), .groups = 'drop') %>%
+  ggplot(aes(x = tmin, y = count, label = 1:28)) +
+  labs(x = 'Minimn))um temperature',
+       y = 'No. of trips',
+       title = 'Trips tend to be low when there is substantial precipitation.') +
+  geom_point() +
+  geom_smooth() +
+  geom_text(hjust = -0.3) +
+  facet_wrap(~substantial_prcp, ncol = 1)

```

Comment:
Our opinions differ about substantial precip haha.

Reply:
how do you define it? i am thinking if we perform a hypothesis testing.

Reply:
It's the amount of precipitation in inches.